cmd/clusterctl/client/rollout_test.go

```diff
@@ -0,0 +1,166 @@
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+)
+
+func Test_clusterctlClient_RolloutRestart(t *testing.T) {
+	type fields struct {
+		client *fakeClient
+	}
+	type args struct {
+		options RolloutRestartOptions
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "do not return error if machinedeployment found",
+			fields: fields{
+				client: fakeClientForRollout(),
+			},
+			args: args{
+				options: RolloutRestartOptions{
+					Kubeconfig: Kubeconfig{Path: "kubeconfig", Context: "mgmt-context"},
+					Resources:  []string{"machinedeployment/md-1"},
+					Namespace:  "default",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "do not return error if all machinedeployments found",
+			fields: fields{
+				client: fakeClientForRollout(),
+			},
+			args: args{
+				options: RolloutRestartOptions{
+					Kubeconfig: Kubeconfig{Path: "kubeconfig", Context: "mgmt-context"},
+					Resources:  []string{"machinedeployment/md-1", "machinedeployment/md-2"},
+					Namespace:  "default",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "return an error is machinedeployment not found",
+			fields: fields{
+				client: fakeClientForRollout(),
+			},
+			args: args{
+				options: RolloutRestartOptions{
+					Kubeconfig: Kubeconfig{Path: "kubeconfig", Context: "mgmt-context"},
+					Resources:  []string{"machinedeployment/foo"},
+					Namespace:  "default",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "return error if one of the machinedeployments is not found",
+			fields: fields{
+				client: fakeClientForRollout(),
+			},
+			args: args{
+				options: RolloutRestartOptions{
+					Kubeconfig: Kubeconfig{Path: "kubeconfig", Context: "mgmt-context"},
+					Resources:  []string{"machinedeployment/md-1", "machinedeployment/md-does-not-exist"},
+					Namespace:  "default",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "return error if unknown resource specified",
+			fields: fields{
+				client: fakeClientForRollout(),
+			},
+			args: args{
+				options: RolloutRestartOptions{
+					Kubeconfig: Kubeconfig{Path: "kubeconfig", Context: "mgmt-context"},
+					Resources:  []string{"foo/bar"},
+					Namespace:  "default",
+				},
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			err := tt.fields.client.RolloutRestart(tt.args.options)
+			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
+				return
+			}
+			g.Expect(err).NotTo(HaveOccurred())
+		})
+	}
+}
+
+func fakeClientForRollout() *fakeClient {
+	core := config.NewProvider("cluster-api", "https://somewhere.com", clusterctlv1.CoreProviderType)
+	infra := config.NewProvider("infra", "https://somewhere.com", clusterctlv1.InfrastructureProviderType)
+	md1 := &clusterv1.MachineDeployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "MachineDeployment",
+			APIVersion: "cluster.x-k8s.io/v1alpha3",
```

Comment:
From the test failures, you may need to update these to v1alpha4. However, I didn't think we needed the APIVersion specified, so I think you can remove this.